source/module_rdmft/rdmft.h

```diff
@@ -0,0 +1,254 @@
+    std::vector<TK> HK_dft_XC;
+
+    hamilt::HS_Matrix_K<TK>* hsk_TV;
+    hamilt::HS_Matrix_K<TK>* hsk_hartree;
+    hamilt::HS_Matrix_K<TK>* hsk_dft_XC;
+    hamilt::HS_Matrix_K<TK>* hsk_exx_XC;
+
+    std::vector<TK> HK_XC;
+    std::vector<TK> HK_local;
+    std::vector< std::vector<TK> > DM_XC_pass;
+    // ModuleDirectMin::ProdStiefelVariable HK_RDMFT_pass;
+    // ModuleDirectMin::ProdStiefelVariable HK_XC_pass;
+
+    ModuleBase::matrix Etotal_n_k;
+    ModuleBase::matrix wfcHwfc_TV;
+    ModuleBase::matrix wfcHwfc_hartree;
+    ModuleBase::matrix wfcHwfc_XC;
+    ModuleBase::matrix wfcHwfc_exx_XC;
+    ModuleBase::matrix wfcHwfc_dft_XC;
+
+    psi::Psi<TK> H_wfc_TV;
+    psi::Psi<TK> H_wfc_hartree;
+    psi::Psi<TK> H_wfc_XC;
+    psi::Psi<TK> H_wfc_exx_XC;
+    psi::Psi<TK> H_wfc_dft_XC;
+
+    // just for temperate. in the future when realize psiDotPsi() without pzgemm_/pdgemm_,we don't need it
+    std::vector<TK> Eij_TV;
+    std::vector<TK> Eij_hartree;
+    std::vector<TK> Eij_XC;
+    std::vector<TK> Eij_exx_XC;
+
+    hamilt::OperatorLCAO<TK, TR>* V_ekinetic_potential = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_nonlocal = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_local = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_hartree = nullptr;
+    // hamilt::OperatorLCAO<TK, TR>* V_XC = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_exx_XC = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_dft_XC = nullptr;
+    hamilt::OperatorLCAO<TK, TR>* V_hartree_XC = nullptr;
+    // bool get_V_local_temp = true;
+
+    Exx_LRI<double>* Vxc_fromRI_d = nullptr;
+    Exx_LRI<std::complex<double>>* Vxc_fromRI_c = nullptr;
+
+    double Etotal = 0.0;
+    double etxc = 0.0;
+    double vtxc = 0.0;
+    bool only_exx_type = false;
+    const int cal_E_type = 1;   // cal_type = 2 just support XC-functional without exx
+
+    void init(Gint_Gamma& GG_in, Gint_k& GK_in, Parallel_Orbitals& ParaV_in, UnitCell& ucell_in,
+                        K_Vectors& kv_in, elecstate::ElecState& pelec_in, LCAO_Orbitals& orb_in, TwoCenterBundle& two_center_bundle_in, std::string XC_func_rdmft_in, double alpha_power_in);
+
+    // update in ion-step and get V_TV
+    void update_ion(UnitCell& ucell_in, ModulePW::PW_Basis& rho_basis_in,
+                        ModuleBase::matrix& vloc_in, ModuleBase::ComplexMatrix& sf_in);
+
+    // update in elec-step
+    // Or we can use rdmft_solver.wfc/occ_number directly when optimizing, so that the update_elec() function does not require parameters.
+    void update_elec(const ModuleBase::matrix& occ_number_in, const psi::Psi<TK>& wfc_in, const Charge* charge_in = nullptr);
+
+    // update occ_number for optimization algorithms that depend on Hamilton
+    void update_occNumber(const ModuleBase::matrix& occ_number_in);
+
+    // // update occ_number for optimization algorithms that depend on Hamilton
+    // void update_wg(const ModuleBase::matrix& wg_in);
+
+    // get the total Hamilton in k-space
+    void cal_Hk_Hpsi();
```

Comment:
Seems these two can be private.

Reply:
OK, I will mark cal_Hk_Hpsi() as private. 

update_occNumber() is an interface reserved for the optimizer and is not used yet, public should be more appropriate.